capstone/src/main/webapp/script.js

```diff
@@ -18,7 +18,11 @@
  * restaurant data
  */
 function getRestaurants() {
-  fetch('/search/restaurants').then(function(response) {
+  const params = new URLSearchParams(window.location.search);
+
+  const queryParam =
+      ((params.get('query') !== null) ? params.get('query') : '');
```

Comment:
I thought the search servlet supported missing parameters. This doesn't necessarily hurt, but it suggests to the reader that there's a vulnerability in the search servlet.

Reply:
The servlet does support missing parameters. It's just that I wanted to be able to "fluently" add the query for now, instead of adding branching logic. I now realize that I can just send the search parameters along for the ride, which might be better than parsing them.